src/rulesets/oas/functions/__tests__/oasDocumentSchema.test.ts

```diff
@@ -189,4 +190,158 @@ describe('oasDocumentSchema', () => {
       ]);
     });
   });
+
+  describe('prepareResults', () => {
+    test('given oneOf error one of which is required $ref property missing, picks only one error', () => {
+      const errors: ErrorObject[] = [
+        {
+          keyword: 'type',
+          dataPath: '/paths/test/post/parameters/0/schema/type',
+          schemaPath: '#/properties/type/type',
+          params: { type: 'string' },
+          message: 'should be string',
+        },
+        {
+          keyword: 'required',
+          dataPath: '/paths/test/post/parameters/0/schema',
+          schemaPath: '#/definitions/Reference/required',
+          params: { missingProperty: '$ref' },
+          message: "should have required property '$ref'",
+        },
+        {
+          keyword: 'oneOf',
+          dataPath: '/paths/test/post/parameters/0/schema',
+          schemaPath: '#/properties/schema/oneOf',
+          params: { passingSchemas: null },
+          message: 'should match exactly one schema in oneOf',
+        },
+      ];
+
+      prepareResults(errors);
+
+      expect(errors).toStrictEqual([
```

Comment:
@P0lip In that case, I wonder if we shouldn't pick the `oneOf` error. It gives a proper direction to work with and doesn't lead the user into fixing a specific branch of the `oneOf`, no?

Reply:
Actually `oneOf` is pretty useless here as we have 2 options for most of the time.
There are a few exceptions, i.e. security components I covered manually, but apart from them we usually deal with a simple case that can be literally expressed as: "either proper value of $ref property". That being said, I strip both `oneOf` and `$ref`, since we are always interested in the first error. Why? Both oas-schema rules operate on resolved content, so it's actually not really `oneOf` anymore - it's always the first member of oneOf we match against.
Does it make sense?

Reply:
@P0lip Yep! Pretty clear! The "already resolved" part indeed sells it.

Could you add that explanation as comments in the error filtering function?